File: ClientApp/src/app/models/Producer.ts
import { Country } from "./Country";
import {Product} from "./Product";
export class Producer{    
    public idProducer?: number;
    public name?: string;
    public Products?: Product[];
    public Country: Country

    constructor(params: Producer = {} as Producer){
        let{
            idProducer = undefined,
            name = undefined,
            Products = [],
            Country = undefined
        } = params;
        this.name = name;
        this.Products = Products;
        this.Country = Country;
    }
}

File: Models/Data/Atributes/Producer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ProductsReviewsAngular.Models
{
    public class Producer
    {
        public int idProducer { get; set; }
        [MaxLength(50)]
        [ConcurrencyCheck]
        public string name { get; set; }
        public string normalizedName { get; set; }
        public virtual ICollection<Product> Products { get; set; }
        public virtual Country Country { get; set; }
    }
}
